### PR TITLE
curvefs client : fix pjdtest problem(create dentry should update pare…

### LINKED ISSUE #689: The pjdtest test failed in 43 use cases
**Describe the bug (描述bug)**
The pjdtest test failed in 43 use cases

**To Reproduce (复现方法)**

**Expected behavior (期望行为)**

**Versions (各种版本)**
OS:
Compiler:
curve-mds:
curve-chunkserver:
curve-snapshotcloneserver:
curve-sdk:
nebd:
curve-nbd:

**Additional context/screenshots (更多上下文/截图)**
image

### PR DESCRIPTION
**Pull request by xu-chaojie**
…nt inode, utimensat, rdev)

### What problem does this PR solve?

Issue Number: close #689 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [x] Relevant documentation/comments is changed or added
- [x] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by xu-chaojie** [comment]
recheck

File: curvefs/src/client/fuse_volume_client.cpp [hunk 1]
```diff
@@ -205,9 +205,12 @@ CURVEFS_ERROR FuseVolumeClient::FuseOpWrite(fuse_req_t req, fuse_ino_t ino,
...
 
-    uint64_t nowTime = TimeUtility::GetTimeofDaySec();
-    inode.set_ctime(nowTime);
-    inode.set_atime(nowTime);
+    struct timespec now;
+    clock_gettime(CLOCK_REALTIME, &now);
+    inode.set_mtime(now.tv_sec);
+    inode.set_mtime_ns(now.tv_nsec);
+    inode.set_ctime(now.tv_sec);
+    inode.set_ctime_ns(now.tv_nsec);
```
Comment (cw123) [inline curvefs/src/client/fuse_volume_client.cpp:213]: old code is ctime and atime
new code is mtime and ctime

File: curvefs/src/metaserver/inode_manager.cpp [hunk 2]
```diff
@@ -34,6 +35,7 @@ MetaStatusCode InodeManager::CreateInode(uint32_t fsId, uint64_t inodeId,
                                          uint32_t gid, uint32_t mode,
                                          FsFileType type,
                                          const std::string &symlink,
+                                         uint32_t rdev,
```
Comment (cw123) [inline curvefs/src/metaserver/inode_manager.cpp:38]: add rdev to log

File: curvefs/proto/metaserver.proto [hunk 3]
```diff
@@ -173,24 +174,27 @@ message S3ChunkInfoList {
...
+    required uint32 mtime_ns = 7;
+    required uint64 atime = 8;
+    required uint32 atime_ns = 9;
+    required uint32 uid = 10;
+    required uint32 gid = 11;
+    required uint32 mode = 12;
+    required uint32 nlink = 13;
+    required FsFileType type = 14;
+    optional string symlink = 15;   // TYPE_SYM_LINK only
+    optional uint32 rdev = 16;
```
Comment (wu-hanqing) [inline curvefs/proto/metaserver.proto:193]: it seems st_rdev in struct stat is `unsigned long`

File: curvefs/proto/metaserver.proto [hunk 4]
```diff
@@ -209,7 +213,8 @@ message CreateInodeRequest {
     required uint32 gid = 7;
     required uint32 mode = 8;
     required FsFileType type = 9;
-    optional string symlink = 10;   // TYPE_SYM_LINK only
+    optional uint32 rdev = 10;
```
Comment (wu-hanqing) [inline curvefs/proto/metaserver.proto:216]: ditto

File: curvefs/src/client/fuse_s3_client.cpp [hunk 5]
```diff
@@ -110,9 +110,12 @@ CURVEFS_ERROR FuseS3Client::FuseOpWrite(fuse_req_t req, fuse_ino_t ino,
...
     }
-    uint64_t nowTime = TimeUtility::GetTimeofDaySec();
-    inode->set_mtime(nowTime);
-    inode->set_ctime(nowTime);
+    struct timespec now;
+    clock_gettime(CLOCK_REALTIME, &now);
+    inode->set_mtime(now.tv_sec);
+    inode->set_mtime_ns(now.tv_nsec);
+    inode->set_ctime(now.tv_sec);
+    inode->set_ctime_ns(now.tv_nsec);
```
Comment (wu-hanqing) [inline curvefs/src/client/fuse_s3_client.cpp:118]: repeated in many places, extract it into a function

File: curvefs/src/client/fuse_volume_client.cpp (hunk 1 above)
Comment (xu-chaojie) [reply curvefs/src/client/fuse_volume_client.cpp:213]: Write modify  mtime is right.

File: curvefs/src/metaserver/inode_manager.cpp [hunk 6]
```diff
@@ -183,6 +197,86 @@ MetaStatusCode InodeManager::UpdateInode(const Inode &inode) {
...
+                   << ", ret = " << MetaStatusCode_Name(ret);
+        return ret;
+    }
+
+    VLOG(1) << "UpdateInodeWhenCreateSubNode success, "
+            << inode.ShortDebugString();
+    return MetaStatusCode::OK;
+}
+
+MetaStatusCode InodeManager::UpdateInodeWhenRemoveSubNode(uint32_t fsId,
```
Comment (wu-hanqing) [inline curvefs/src/metaserver/inode_manager.cpp:240]: it seems these two functions only differ in update inode nlink, previous one is increasing and this one is decreasing.

File: curvefs/src/metaserver/inode_manager.cpp [hunk 7]
```diff
@@ -98,18 +100,26 @@ MetaStatusCode InodeManager::CreateRootInode(uint32_t fsId, uint32_t uid,
...
+    inode->set_rdev(rdev);
+
+    struct timespec now;
+    clock_gettime(CLOCK_REALTIME, &now);
+    inode->set_mtime(now.tv_sec);
+    inode->set_mtime_ns(now.tv_nsec);
+    inode->set_atime(now.tv_sec);
+    inode->set_atime_ns(now.tv_nsec);
+    inode->set_ctime(now.tv_sec);
+    inode->set_ctime_ns(now.tv_nsec);
```
Comment (wu-hanqing) [inline curvefs/src/metaserver/inode_manager.cpp:121]: also repeated

File: curvefs/src/metaserver/partition.cpp [hunk 8]
```diff
@@ -57,14 +57,40 @@ MetaStatusCode Partition::CreateDentry(const Dentry& dentry) {
...
+MetaStatusCode Partition::InsertDentry(const Dentry& dentry) {
+    if (!IsInodeBelongs(dentry.fsid(), dentry.parentinodeid())) {
+        return MetaStatusCode::PARTITION_ID_MISSMATCH;
+    }
+    MetaStatusCode ret = dentryManager_->CreateDentry(dentry);
+    if (MetaStatusCode::IDEMPOTENCE_OK == ret) {
+        return MetaStatusCode::OK;
+    } else {
+        return ret;
+    }
```
Comment (wu-hanqing) [inline curvefs/src/metaserver/partition.cpp:80]: there is no much difference with `CreateDentry`, use another flag to control whether do update

File: curvefs/src/metaserver/inode_manager.cpp (hunk 2 above)
Comment (xu-chaojie) [reply curvefs/src/metaserver/inode_manager.cpp:38]: done

File: curvefs/proto/metaserver.proto (hunk 3 above)
Comment (xu-chaojie) [reply curvefs/proto/metaserver.proto:193]: done

File: curvefs/proto/metaserver.proto (hunk 4 above)
Comment (xu-chaojie) [reply curvefs/proto/metaserver.proto:216]: done

File: curvefs/src/metaserver/inode_manager.cpp (hunk 6 above)
Comment (xu-chaojie) [reply curvefs/src/metaserver/inode_manager.cpp:240]: done

File: curvefs/src/metaserver/partition.cpp (hunk 8 above)
Comment (xu-chaojie) [reply curvefs/src/metaserver/partition.cpp:80]: done

**Review by wu-hanqing — APPROVED** [review]

**Review by cw123 — APPROVED** [review]

**Comment by xu-chaojie** [comment]
recheck

**Comment by xu-chaojie** [comment]
recheck

**Pull request merged**